tests/Http/Api/PostApiTest.php

```diff
@@ -125,6 +125,99 @@ public function testCreatingAPost()
         ]);
     }
 
+    /**
+     * Test that a POST request to /posts with an existing post creates an additional new photo post.
+     *
+     * @return void
+     */
+    public function testCreatingMultiplePosts()
+    {
+        $signup = factory(Signup::class)->create();
+        $quantity = $this->faker->numberBetween(10, 1000);
+        $caption = $this->faker->sentence;
+
+        // Mock the Blink API call.
+        $this->mock(Blink::class)->shouldReceive('userSignupPost');
+
+        // Create the post!
+        $response = $this->withRogueApiKey()->json('POST', 'api/v2/posts', [
+            'northstar_id'     => $signup->northstar_id,
+            'campaign_id'      => $signup->campaign_id,
+            'campaign_run_id'  => $signup->campaign_run_id,
+            'quantity'         => $quantity,
+            'why_participated' => $this->faker->paragraph,
+            'num_participants' => null,
+            'caption'          => $caption,
+            'source'           => 'phpunit',
+            'remote_addr'      => $this->faker->ipv4,
+            'file'             => UploadedFile::fake()->image('photo.jpg'),
+            'crop_x'           => 0,
+            'crop_y'           => 0,
+            'crop_width'       => 100,
+            'crop_height'      => 100,
+            'crop_rotate'      => 90,
+        ]);
+
+        $response->assertStatus(200);
```

Comment:
Now that I think about it, shouldn't all of these codes be `201`? When I `dd` the response in the code, it is `201` since these are all created but for some reason the tests return a `200` status code...

Reply:
oh wait now I see - it checks to see if the [signup is created or not](https://github.com/DoSomething/rogue/blob/master/app/Http/Controllers/Three/PostsController.php#L114) and [determines the status code based on that](https://github.com/DoSomething/rogue/blob/master/app/Http/Controllers/Three/PostsController.php#L123) - we might want to change this (I know we talked about this before) since this is a little misleading...

Reply:
yea +1 it really does need to be refactored.